dev-client/src/hooks/userDeletionRequest.ts

```diff
@@ -27,13 +27,12 @@ const PREFERENCES_VALUE = 'true';
 export const useUserDeletionRequests = () => {
   const dispatch = useDispatch();
   const {data: user} = useSelector(state => state.account.currentUser);
-  const [isPending, setIsPending] = useState(
-    user!.preferences[PREFERENCES_KEY] === PREFERENCES_VALUE,
-  );
+  const isSaving = useSelector(state => state.account.preferences.saving);
```

Comment:
The state here was unnecessary; we can just look at the user's preference value since it will get updated when the dispatched request completes. (We now disable the action button while the request is pending for a smoother user experience.)